app/src/main/java/org/mozilla/focus/browser/BrowserFragment.kt

```diff
@@ -183,10 +184,16 @@ class BrowserFragment : EngineViewLifecycleFragment(), Session.Observer {
             onNavigationEvent = this@BrowserFragment.onNavigationEvent
             navigationStateProvider = NavigationStateProvider()
             visibility = overlayVisibleCached ?: View.GONE
+            onPreSetVisibilityListener = { isVisible ->
```

Comment:
@liuche You removed this in https://github.com/mozilla-mobile/firefox-tv/commit/e1570a04b68aaac0c4000b727d19e0a84b7a9bde, but this is actually necessary because when we refresh(), YT resets the JS cached var;. We need another entry point to request focus on YT webView by injecting JS body again after reset()

Please see:
https://github.com/mozilla-mobile/firefox-tv/issues/393
https://github.com/mozilla-mobile/firefox-tv/issues/682
https://github.com/mcomella/firefox-tv/commit/2b4acfe1d4dc249f2d435522f004d3d8c44c7f95#diff-47491f34b37c83915496a11a48753fd7